packages/types/src/pagination.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Expected type definition of a paginator.
+ */
+export type Paginator<T> = AsyncGenerator<T, T, unknown>;
+
+/**
+ * Expected paginator configuration passed to an operation. Services will extend
+ * this interface definition and may type client further.
+ */
+export interface PaginationConfiguration {
+  client: unknown;
```

Comment:
Should this by [`Client`](https://github.com/aws/aws-sdk-js-v3/blob/ffe353b8c66b5b8d9777aa1c26b72042b1af5285/packages/types/src/client.ts#L30)? Like `Client<any, any, any>`?

Reply:
yes, I think it should. good callout

Reply:
Updated

Reply:
Can you use the Client interface from `types` package? https://github.com/aws/aws-sdk-js-v3/blob/ffe353b8c66b5b8d9777aa1c26b72042b1af5285/packages/types/src/client.ts#L30
This will avoid this package has dependency over `smithy-client` package

Reply:
done.  Nice catch